File: 프로그래머스/lv2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    clothes_tb = {}
    
    for cloth_name, cloth_type in clothes:
        clothes_tb.setdefault(cloth_type, [])
        clothes_tb[cloth_type].append(cloth_name)
    
    for cloth_type in clothes_tb.keys():
        answer *= len(clothes_tb[cloth_type])+1
    
    return answer-1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

/*
종류별로 갯수를 세서 조합의 수를 구한다.
*/

class Solution {
    public int solution(String[][] clothes) {
        Map<String, Integer> cMap = new HashMap<>();
        
        for (String[] c : clothes) {
            cMap.put(c[1], cMap.getOrDefault(c[1], 0) + 1);
        }
        
        int answer = 1;
        
        for (Integer v : cMap.values()) {
            answer *= v + 1;
        }
        
        return answer - 1;
    }
}